src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -33,33 +38,59 @@ typedef struct gavl_tree_node {
     const void *val;
 } gavl_tree_node_s;
 
+typedef struct gavl_tree_stack {
+    gavl_tree_node_s *stack[MAX_STACK_SIZE];
+    int stack_sp;
+} gavl_tree_stack_s;
+
 typedef struct gavl_tree {
     gavl_tree_node_s *root;
     void (*gavl_free_fn) (void *);
+    gavl_tree_stack_s node_stack;
+    gavl_tree_node_s *cur_node;
```

Comment:
Can you add a short comment explaining what `node_stack` is used for?

Reply:
Is access to `node_stack` thread safe? What happens if two threads call gavl concurrently?

Reply:
> Can you add a short comment explaining what `node_stack` is used for? Same for `cur_node`. It is not straightforward why you need them in the tree structure.

ok



> Is access to `node_stack` thread safe? What happens if two threads call gavl concurrently?

we have discussed thread safety is provided by MPICH, so all functions in gavl tree are unsafe.

Reply:
The comment for stack seems misleading. It is used at insertion or deletion, but not search. Besides, I do not know if "restore the trace" is right. I guess you want to say rebalance the tree. How about change the comment to 
```
Internal stack structure. Used to track the traversing trace for tree rebalance
at node insertion or deletion. cur_node points to the starting node of tree rebalance.
```